scipy/optimize/_qap.py

```diff
@@ -655,7 +690,9 @@ def _quadratic_assignment_2opt(A, B, maximize=False, rng=None,
     partial_guess = np.atleast_2d(partial_guess).astype(int)
 
     msg = None
-    if partial_guess.shape[0] > A.shape[0]:
+    if A.shape != B.shape:
+        msg = "`A` and `B` matrices must be of equal size"
```

Comment:
Removed from common input validation, so looks good. Curious, though - could 2-opt be used on the padded matrices? (I ask without having looked into what exactly is being done.)

Reply:
no, I don't believe this padding scheme would work on 2-opt